site/app/libraries/NotificationFactory.php

```diff
@@ -234,6 +234,16 @@ public function sendEmails(array $emails, bool $force_secondary = false, bool $s
         $current_user = $this->core->getUser();
         $flattened_emails = [];
         foreach ($emails as $email) {
+            if ($email->getEmailAddress() !== null && $email->getEmailAddress() !== '') {
```

Comment:
Could we just use !empty() here?

Reply:
I had that, but Will wanted me to change it to this to be more explicit. I think I will change it back though, as I don't think Will understood that I was also considering the null case in my logic (he recommended `$email->getEmailAddress() !== ''` originally).

Reply:
@williamjallen

Reply:
`getEmailAddress()` returns a string, so we can be more explicit by checking against an empty string.  `empty()` provides a pretty loose comparison, so it should be avoided if possible.